modules/photons_control/clean.py

```diff
@@ -0,0 +1,189 @@
+    reference = task.provides_reference(special=True)
+
+    async def execute_task(self, **kwargs):
+        async def gen(reference, sender, **kwargs):
+            ps = sender.make_plans("capability")
+            async for serial, _, info in sender.gatherer.gather(ps, reference, **kwargs):
+                if info["cap"].has_hev:
+                    yield LightMessages.GetHevCycleConfiguration(target=serial)
+
+        async for pkt in self.target.send(FromGenerator(gen), self.reference):
+            print(pkt.serial)
+            indication = "enabled" if pkt.indication else "disabled"
+            print(f"   Cycle time: {pkt.duration_s}")
+            print(f"   Indication: {indication} (raw: {pkt.indication})")
+            print("")
+
+
+def StartStopCleanCycle(enable=None, duration=None, reference=None, **kwargs):
+    """
+    Returns a valid message that will either start or stop the cleaning (HEV)
+    cycle of devices used against it.
+
+    When a cycle is started, if the duration=0 or is not provided, the default
+    duration will be used.
+
+    For example:
+
+    .. code-block:: python
+
+        await target.send(CleanCycle(), ["d073d5000001", "d073d5000001"])
+    """
+
+    async def gen(ref, sender, **kwargs):
+        r = ref if reference is None else reference
+        d = 0 if duration is NotSpecified else duration
+
+        plans = sender.make_plans("capability")
+        async for serial, _, info in sender.gatherer.gather(plans, r, **kwargs):
+            if info["cap"].has_hev:
+                yield LightMessages.SetHevCycle(
+                    enable=enable,
+                    res_required=False,
+                    ack_required=True,
+                    duration_s=d,
+                    target=serial,
+                )
+
+    return FromGenerator(gen)
+
+
+@task
+class start_clean_cycle(task.Task):
+    """
+    Start a cleaning cycle on an HEV-capable bulb
+
+    ``target:start_clean_cycle match:cap=hev -- '{"duration": 7200}'``
+
+    It  takes a ``duration`` field that is the seconds of the duration. If not
+    provided, or set to 0, the default duration will be used.
+    """
+
+    target = task.requires_target()
+    reference = task.provides_reference(special=True)
+
+    async def execute_task(self, **kwargs):
+        options = self.photons_app.extra_as_json
+        options = sb.set_options(
+            enable=sb.defaulted(sb.boolean(), True), duration=sb.integer_spec()
+        ).normalise(Meta.empty(), options)
+
+        await self.target.send(StartStopCleanCycle(**options), self.reference)
+
+
+@task
+class stop_clean_cycle(task.Task):
+    """
+    Stop a cleaning cycle on an HEV-capable bulb
+
+    ``target:stop_clean_cycle match:cap=hev``
+
+    """
+
+    target = task.requires_target()
+    reference = task.provides_reference(special=True)
+
+    async def execute_task(self, **kwargs):
+        options = self.photons_app.extra_as_json
+        options = sb.set_options(
```

Comment:
so if this set_options is in the special message object, then you don't need to validate it everywhere you use the special message object